phpBB/phpbb/content_visibility.php

```diff
@@ -143,12 +143,42 @@ public function get_count($mode, $data, $forum_id)
 	*/
 	public function get_visibility_sql($mode, $forum_id, $table_alias = '')
 	{
+		$where_sql = '';
+		
+		$get_visibility_sql_overwrite = false;
+		/**
+		* Allow changing the result of calling get_visibility_sql
+		*
+		* @event core.phpbb_content_visibility_get_forums_visibility_before
+		* @var	string		where_sql						Extra visibility conditions. It must end with either an SQL "AND" or an "OR"
+		* @var	string		mode							Either "topic" or "post" depending on the query this is being used in
+		* @var	array		forum_id						The forum ids which the post/topic is from
+		* @var	string		table_alias						Table alias to prefix in SQL queries
+		* @var	mixed		get_visibility_sql_overwrite	If a string, forces the function to return get_forums_visibility_sql_overwrite after executing the event
+		* 													If false, get_visibility_sql continues normally
+		* 													It must be either boolean or string
+		* @since 3.1.4-RC1
+		*/
+		$vars = array(
+			'where_sql',
+			'mode',
+			'forum_id',
+			'table_alias',
+			'get_visibility_sql_overwrite',
+		);
```

Comment:
I feel like if this went back in time. I had changed all that O_O.
I wonder if it was git stash that borked it when I made a pull before 
the commit...
I'll recheck everything again

On 09-02-2015 08:09, Joas Schilling wrote:
>
> In phpBB/phpbb/content_visibility.php 
> <https://github.com/phpbb/phpbb/pull/3384#discussion_r24313361>:
>
> > +		* @var	string		where_sql						Extra visibility conditions. It must end with either an SQL "AND" or an "OR"
> > +		* @var	string		mode							Either "topic" or "post" depending on the query this is being used in
> > +		* @var	array		forum_id						The forum ids which the post/topic is from
> > +		* @var	string		table_alias						Table alias to prefix in SQL queries
> > +		* @var	mixed		get_visibility_sql_overwrite	If a string, forces the function to return get_forums_visibility_sql_overwrite after executing the event
> > +		* 													If false, get_visibility_sql continues normally
> > +		* 													It must be either boolean or string
> > +		* @since 3.1.4-RC1
> > +		*/
> > +		$vars = array(
> > +			'where_sql',
> > +			'mode',
> > +			'forum_id',
> > +			'table_alias',
> > +			'get_visibility_sql_overwrite',
> > +		);
>
> remove |forums|
>
> |LogicException: The event 'core.phpbb_content_visibility_get_forums_visibility_before' from file 'phpbb/content_visibility.php:204' already exists in file 'phpbb/content_visibility.php'
> |
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/phpbb/phpbb/pull/3384/files#r24313361>.
>